src/nomad_simulations/model_system.py

```diff
@@ -967,6 +977,25 @@ def normalize(self, archive, logger) -> None:
         if is_not_representative(self, logger):
             return
 
+        # Traversing the System Hierarchy
+        # TODO consider moving to utils and passing generic set of functions to execute at each branch level
+        def set_branch_composition(system, subsystems, atom_labels):
+            if subsystems is None:
+                subsystem_labels = [np.array(atom_labels)[system.get("atom_indices")]]
+            else:
+                subsystem_labels = [subsystem.get("branch_label") for subsystem in subsystems]
+            system.composition_formula = get_composition(subsystem_labels)
+
+        def traverse_system_recurs(system, atom_labels):
+            subsystems = system.get("model_system")
+            set_branch_composition(system, subsystems, atom_labels)
+            if subsystems:
+                for subsystem in subsystems:
+                    traverse_system_recurs(subsystem, atom_labels)
+
+        atom_labels = [atom.chemical_symbol for atom in self.cell[0].atoms_state]
```

Comment:
I think you need some error handling here, to consider if `self.cell` exists and if so, if `atoms_state` exist. Something like:
```python
atom_labels = [atom.chemical_symbol for atom in self.cell[0].atoms_state if self.cell is not None and self.cell.atoms_state is not None]
```